flink-ml-core/src/test/java/org/apache/flink/ml/util/QuantileSummaryTest.java

```diff
@@ -0,0 +1,135 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.util;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/** Tests {@link QuantileSummary}. */
+public class QuantileSummaryTest {
+
+    private List<double[]> datasets;
+
+    @Before
+    public void prepare() {
+        double[] increasing = IntStream.range(0, 100).mapToDouble(x -> x).toArray();
+        double[] decreasing = IntStream.range(0, 100).mapToDouble(x -> 99 - x).toArray();
+        double[] negatives = IntStream.range(-100, 0).mapToDouble(x -> x).toArray();
+
+        datasets = new ArrayList<>(Arrays.asList(increasing, decreasing, negatives));
+    }
+
+    private QuantileSummary buildSummary(double[] data, double epsilon) {
+        QuantileSummary summary = new QuantileSummary(epsilon);
+        for (int i = 0; i < data.length; i++) {
+            summary = summary.insert(data[i]);
+        }
+        return summary.compress();
+    }
+
+    private void checkQuantiles(double[] data, double[] percentiles, QuantileSummary summary) {
+        if (data.length == 0) {
+            assertNull(summary.query(percentiles));
+        } else {
+            double[] quantiles = summary.query(percentiles);
+            IntStream.range(0, percentiles.length)
+                    .forEach(
+                            i ->
+                                    validateApproximation(
+                                            quantiles[i], data, percentiles[i], summary));
+        }
+    }
+
+    private void validateApproximation(
+            double approx, double[] data, double percentile, QuantileSummary summary) {
+        double rank =
+                Math.ceil(
+                        (Arrays.stream(data).filter(x -> x <= approx).count()
+                                        + Arrays.stream(data).filter(x -> x < approx).count())
+                                / 2.0);
+        double lower = Math.floor((percentile - summary.getRelativeError()) * data.length);
+        double upper = Math.ceil((percentile + summary.getRelativeError()) * data.length);
+        String errMessage =
+                String.format(
+                        "Rank not in [%s, %s], percentile: %s, approx returned: %s",
+                        lower, upper, percentile, approx);
+        assertTrue(errMessage, rank >= lower);
+        assertTrue(errMessage, rank <= upper);
```

Comment:
Would it be better to use the following method? `assertEquals(String errMessage, double expected, double actual, double epsilon)`